src/test/java/liuni/KeyHandlerTests.java

```diff
@@ -0,0 +1,152 @@
+package liuni;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.xml.sax.SAXException;
+import twitter4j.Twitter;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class KeyHandlerTests {
+
+    private String twitter4jFileName;
+
+    @Mock private Twitter twitter;
+
+    @InjectMocks KeyHandler keyHandler;
+
+    @Before
+    public void setUp() {
+        twitter4jFileName = "twitter4j.properties";
+
+        MockitoAnnotations.initMocks(this);
+        keyHandler = new KeyHandler();
+        keyHandler.setTwitter(twitter);
+
+        File HCKeysFile = new File("hardcoded_keys.xml");
+        if (!HCKeysFile.exists()) {
```

Comment:
if you were to stick to file creation, I'd recommend to only create if needed and not on all tests.  You've got many tests that successively delete the file.  I'd still go with my later comment however with the test framework.